notebooks/concepts/imviz_compass_mpl.ipynb

```diff
@@ -0,0 +1,251 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "9a8d6eac-6626-415d-ade4-3f986cf18a69",
+   "metadata": {},
+   "source": [
+    "# Imviz compass visualization using Matplotlib\n",
+    "\n",
+    "This concept notebook visualizes the compass algorithm that Imviz will use. For simplicity, visualization is done using Matplotlib.\n",
+    "\n",
+    "These examples are also in the unit tests for `wcs_utils.py`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e571e723",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import gwcs\n",
+    "import numpy as np\n",
+    "from astropy import units as u\n",
+    "from astropy.coordinates import ICRS\n",
+    "from astropy.modeling import models\n",
+    "from astropy.wcs import WCS\n",
+    "from gwcs import coordinate_frames as cf\n",
+    "\n",
+    "from jdaviz.configs.imviz import wcs_utils"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "fd58f8ef",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%matplotlib inline"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3779ae77-97fa-49db-85e5-7e8213c0cd30",
+   "metadata": {},
+   "source": [
+    "## Case 1: Dummy WCS with 10x10 image"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ae33669d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "a = np.arange(100).reshape((10, 10))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c334e59e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "w = WCS()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "20605e33",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "wcs_utils.draw_compass_mpl(a, w)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8999c3b9-2bf3-4b6a-89e9-5f6ec843a73e",
+   "metadata": {},
+   "source": [
+    "## Case 2: Real WCS with no rotation nor distortion\n",
+    "\n",
+    "We reuse the array from above but WCS from https://learn.astropy.org/tutorials/celestial_coords1.html example."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3253dd7f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "w = WCS({'CTYPE1': 'RA---TAN',\n",
+    "         'CUNIT1': 'deg',\n",
+    "         'CDELT1': -0.0002777777778,\n",
+    "         'CRPIX1': 1,\n",
+    "         'CRVAL1': 337.5202808,\n",
+    "         'NAXIS1': 10,\n",
+    "         'CTYPE2': 'DEC--TAN',\n",
+    "         'CUNIT2': 'deg',\n",
+    "         'CDELT2': 0.0002777777778,\n",
+    "         'CRPIX2': 1,\n",
+    "         'CRVAL2': -20.833333059999998,\n",
+    "         'NAXIS2': 10})"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d2e73414",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "wcs_utils.draw_compass_mpl(a, w)"
```

Comment:
| This PR | Ginga |
| --- | --- |
| ![dummy_wcs_learnastropy_imviz](https://user-images.githubusercontent.com/2090236/144470583-859bb840-e2ba-4ae0-a406-6e06e48c9c2a.jpg) | ![dummy_wcs_learnastropy_ginga](https://user-images.githubusercontent.com/2090236/144470597-4da5343a-3088-4345-a2d3-6f7a11148c6c.jpg) |